probe/probe.go

```diff
@@ -302,3 +329,347 @@ func probeNode(c *k8s.Client) {
+		}
+	}
+
+	return kubearmorContainerCount, kubearmorImageVersion
+
+}
+
+func getKubeArmorRelayContainerCountAndImageVersion(c *k8s.Client) (int, string) {
+	// KubeArmor Relay Pod
+	relayContainerCount := 0
+	relayImageVersion := ""
+	relayPods, err := c.K8sClientset.CoreV1().Pods("kube-system").List(context.Background(), metav1.ListOptions{
+		LabelSelector: "kubearmor-app=kubearmor-relay",
+	})
+	if err != nil {
+		//log.Println("error occured when getting kubearmor relay pods")
+		return relayContainerCount, relayImageVersion
+	}
+
+	if len(relayPods.Items) > 0 {
+		relayImageVersion = relayPods.Items[0].Spec.Containers[0].Image
+		for _, relayPodItem := range relayPods.Items {
+			relayContainerCount += len(relayPodItem.Spec.Containers)
+
+		}
+	}
+	return relayContainerCount, relayImageVersion
+}
+
+func probeRunningKubeArmorNodes(c *k8s.Client) error {
+
+	// // KubeArmor Nodes
+	nodes, err := c.K8sClientset.CoreV1().Nodes().List(context.Background(), metav1.ListOptions{})
+	if err != nil {
+		log.Println("error getting nodes", err)
+		return err
+	}
+
+	if len(nodes.Items) > 0 {
+
+		for i, item := range nodes.Items {
+			_, err := boldWhite.Printf("Node %d : \n", i+1)
+			if err != nil {
+				color.Red(" Error while printing")
+			}
+			err = readDataFromKubeArmor(c, item.Name)
+			if err != nil {
+				return err
+			}
+		}
+	} else {
+		fmt.Println("No kubernetes environment found")
+	}
+	return nil
+}
+
+func readDataFromKubeArmor(c *k8s.Client, nodeName string) error {
+	var kd *KubeArmorProbeData
+
+	srcPath := "/tmp/karmorProbeData.cfg"
+	pods, err := c.K8sClientset.CoreV1().Pods("kube-system").List(context.Background(), metav1.ListOptions{
+		LabelSelector: "kubearmor-app=kubearmor",
+		FieldSelector: "spec.nodeName=" + nodeName,
+	})
+	if err != nil {
+		log.Println("error occured while getting kubeArmor pods", err)
+		return err
+	}
+	reader, outStream := io.Pipe()
+	cmdArr := []string{"cat", srcPath}
+	req := c.K8sClientset.CoreV1().RESTClient().
+		Get().
+		Namespace("kube-system").
+		Resource("pods").
+		Name(pods.Items[0].Name).
+		SubResource("exec").
+		VersionedParams(&corev1.PodExecOptions{
+			Container: pods.Items[0].Spec.Containers[0].Name,
+			Command:   cmdArr,
+			Stdin:     true,
+			Stdout:    true,
+			Stderr:    true,
+			TTY:       false,
+		}, scheme.ParameterCodec)
+	exec, err := remotecommand.NewSPDYExecutor(c.Config, "POST", req.URL())
+
+	if err != nil {
+		return err
+	}
+	go func() {
+		defer outStream.Close()
+		err = exec.Stream(remotecommand.StreamOptions{
+			Stdin:  os.Stdin,
+			Stdout: outStream,
+			Stderr: os.Stderr,
+			Tty:    false,
+		})
+	}()
+	buf, err := io.ReadAll(reader)
+	if err != nil {
+		log.Println("an error occured when reading file", err)
+		return err
+	}
+	var json = jsoniter.ConfigCompatibleWithStandardLibrary
+	err = json.Unmarshal(buf, &kd)
+	if err != nil {
+		log.Println("an error occured when parsing file", err)
+		return err
+	}
+	printKubeArmorProbeOutput(kd)
```

Comment:
```suggestion
	printKubeArmorProbeOutput(buf)
```
Perhaps let's send the buf to print output and return error, since it's being repeated across both the functions.